crates/assists/src/handlers/inline_function.rs

```diff
@@ -0,0 +1,96 @@
+use ast::make;
+use hir::{HasSource, HasVisibility, ModuleDef, PathResolution};
+use syntax::{
+    ast::{self, edit::AstNodeEdit, ArgListOwner},
+    AstNode,
+};
+
+use crate::{
+    assist_context::{AssistContext, Assists},
+    AssistId, AssistKind,
+};
+
+// Assist: inline_function
+//
+// Inlines a function body.
+//
+// ```
+// fn add(a: u32, b: u32) -> u32 { a + b }
+// fn main() {
+//     let x = add<|>(1, 2);
+// }
+// ```
+// ->
+// ```
+// fn add(a: u32, b: u32) -> u32 { a + b }
+// fn main() {
+//     let x = {
+//         let a = 1;
+//         let b = 2;
+//         a + b
+//     };
+// }
+// ```
+pub(crate) fn inline_function(acc: &mut Assists, ctx: &AssistContext) -> Option<()> {
+    let path_expr: ast::PathExpr = ctx.find_node_at_offset()?;
+    let call = path_expr.syntax().parent().and_then(ast::CallExpr::cast)?;
+    let path = path_expr.path()?;
+
+    let function = match ctx.sema.resolve_path(&path)? {
+        PathResolution::Def(ModuleDef::Function(f)) => f,
+        _ => return None,
+    };
+
+    let current_scope = ctx.sema.scope(call.syntax());
+    let current_module = current_scope.module()?;
+
+    if !function.is_visible_from(ctx.db(), current_module) {
+        // The function isn't accessible from here so we can't inline it
+        return None;
+    }
```

Comment:
I've added a test case ([here](https://github.com/rust-analyzer/rust-analyzer/pull/7131/files#diff-7631a83bd6e299dfaee18bc775747868f5e2af182615ffa5b88084c91ab327d2R170)), but I agree that it may not be 100% necessary. 

You can already ctrl-click to jump to an inaccessible function's source code, so there doesn't seem to be a need to enforce privacy here... I think I'll remove the check.

Reply:
I've removed this privacy check. It didn't really provide benefit and this simplifies the code.